test/blackbox-tests/test-cases/coq/duplicate-theory.t/run.t

```diff
@@ -0,0 +1,17 @@
+Duplicate theories should be caught by Dune:
+
+  $ cat > dune << EOF
+  > (coq.theory
+  >  (name foo))
+  > 
+  > (coq.theory
+  >  (name foo))
+  > EOF
+
+BUG The directory target is found before the theory
+
+  $ dune build
+  Error: the following both define the directory target: _build/default/foo.tex
+  - dune:1
```

Comment:
That's because laziness in rule creation and coqdoc not forcing the Coq lib db to be built.

Reply:
I'm not sure. coqdoc rules are generated when we generate the rules for the theory. The issue here is that we have to pre-declare directory targets which due to lazyness means we have identical dir targets.

Reply:
I guess we can reorder and get a better error message? Or just force the Coq.Lib.db in the coqdoc rules.

Reply:
Again it's not the coqdoc rules, its the declaration of the directory target. The coqdoc rules already force the database.

Reply:
Ahh, I see what you mean. Not sure what can be done for it.